packages/block-library/src/navigation/edit/index.js

```diff
@@ -495,6 +528,13 @@ function Navigation( {
 											setRef( id );
 											onClose();
 										} }
+										onSelectClassicMenu={ ( menu ) => {
+											onClose();
```

Comment:
Calling `onClose` here allows the dropdown to close and the block UI to update whilst the async conversion process happens.

Reply:
Need to be a little careful about focus management here.